### PR TITLE
Refactor removegroup

### PR DESCRIPTION
**Pull request by yash-chowdhary**
Refactor `renamegroup` to `deletegroup` in functional and test code.

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -746,48 +746,48 @@ _Figure 32. Sequence Diagram for viewing tasks of the currently logged-in member
...
+=== Delete Group Mechanism
 ==== Current Implementation
 
-The `removegroup` mechanism is facilitated by the `RemoveGroupCommand` class. It allows `Exco` members to remove a group from `Club Connect`. The group of all members part of the group that is to be deleted will be changed to the default group - `member`.
+The `deletegroup` mechanism is facilitated by the `DeleteGroupCommand` class. It allows `Exco` members to delete a group from `Club Connect`. The group of all members part of the group that is to be deleted will be changed to the default group - `member`.
 
-The `RemoveGroupCommand` extends from `UndoableCommand` as it is an undoable command. Figure 33 below depicts the UML representation of the `Command`.
+The `DeleteGroupCommand` extends from `UndoableCommand` as it is an undoable command. Figure 33 below depicts the UML representation of the `Command`.
 
 image::RemoveGroupCommandUML.png[width="200" height="200"]
```
Comment (MuhdNurKamal) [inline docs/DeveloperGuide.adoc]: Might want to change name of the image as well

Comment (MuhdNurKamal) [inline docs/DeveloperGuide.adoc]: Might want to rename image as well

File: src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java [hunk 2]
```diff
@@ -6,32 +6,32 @@
 import java.util.stream.Stream;
```
Comment (MuhdNurKamal) [inline src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java:7]: Rename this class

File: src/main/java/seedu/club/logic/parser/ClubBookParser.java [hunk 3]
```diff
@@ -269,10 +269,10 @@ private boolean isSelectCommand(String commandWord) {
     }
 
     /**
-     * Returns true if {@code commandWord} matches any of RemoveGroupCommand's aliases
+     * Returns true if {@code commandWord} matches any of DeleteGroupCommand's aliases
      */
     private boolean isRemoveGroupCommand(String commandWord) {
```
Comment (MuhdNurKamal) [inline src/main/java/seedu/club/logic/parser/ClubBookParser.java]: Rename method

File: src/test/java/seedu/club/model/ModelManagerTest.java [hunk 4]
```diff
@@ -99,24 +99,24 @@ public void removeGroup_mandatoryGroup_modelUnchanged() throws Exception {
...
         try {
-            modelManager.removeGroup(new Group(MANDATORY_GROUP));
+            modelManager.deleteGroup(new Group(MANDATORY_GROUP));
         } catch (GroupCannotBeRemovedException e) {
             assertEquals(new ModelManager(clubBook, userPrefs), modelManager);
         }
     }
 
     @Test
     public void removeGroup_atLeastOneMemberInGroup_groupRemoved() throws Exception {
```
Comment (MuhdNurKamal) [inline src/test/java/seedu/club/model/ModelManagerTest.java]: rename method name

**Review by MuhdNurKamal — COMMENTED** [review]
Nice, just missed a few lines

File: src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java (hunk 2 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java]: which one? IllegalValueException ?

File: src/main/java/seedu/club/logic/parser/ClubBookParser.java (hunk 3 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/logic/parser/ClubBookParser.java]: 👍

File: src/test/java/seedu/club/model/ModelManagerTest.java (hunk 4 above)
Comment (yash-chowdhary) [reply src/test/java/seedu/club/model/ModelManagerTest.java]: 👍

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (yash-chowdhary) [reply docs/DeveloperGuide.adoc]: ok 👍

File: src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java (hunk 2 above)
Comment (yash-chowdhary) [reply src/main/java/seedu/club/logic/parser/RemoveGroupCommandParser.java:7]: done

**Pull request merged**